spec/morandi_spec.rb

```diff
@@ -300,6 +300,72 @@
     end
   end
 
+  describe 'when applying shrink to fit' do
+    let(:cropped_width) { 1767 }
+    let(:cropped_height) { 1414 }
+
+    let(:options) do
+      {
+        'crop' => [0, -172.9338582677165, cropped_width, cropped_height],
+        'output.width' => cropped_width,
+        'output.height' => cropped_height
+      }
+    end
+
+    it 'should maintain the target size' do
+      process_image
+
+      expect(File).to exist(file_out)
+      expect(processed_image_type).to eq('jpeg')
+      expect(processed_image_width).to eq(cropped_width)
+      expect(processed_image_height).to eq(cropped_height)
+    end
+
+    it 'should when given a print size of 400 by 325 half the size of the input image' do
+      shrink_to_fit_options = { 'crop' => [10, -10, original_image_width, original_image_height], 'output.width' => 400, 'output.height' => 325 }
+
+      Morandi.process(file_in, shrink_to_fit_options, file_out)
```

Comment:
There's a `subject` at the top:
```
  subject(:process_image) do
    Morandi.process(file_in, options, file_out)
  end
```

So, if we set the `let(:options) { ... }`, then calling `process_image` will be much more in-keeping.

Reply:
let(:options) is only valid at a context or describe level - it objects when placed within an it block